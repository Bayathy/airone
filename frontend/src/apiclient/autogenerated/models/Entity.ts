/* tslint:disable */
/* eslint-disable */
/**
 *
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 0.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from "../runtime";
/**
 *
 * @export
 * @interface Entity
 */
export interface Entity {
  /**
   *
   * @type {number}
   * @memberof Entity
   */
  readonly id: number;
  /**
   *
   * @type {string}
   * @memberof Entity
   */
  name: string;
  /**
   *
   * @type {string}
   * @memberof Entity
   */
  note: string;
  /**
   *
   * @type {number}
   * @memberof Entity
   */
  status?: number;
  /**
   *
   * @type {boolean}
   * @memberof Entity
   */
  readonly isToplevel: boolean;
  /**
   *
   * @type {Array<{ [key: string]: any; }>}
   * @memberof Entity
   */
  readonly attrs: Array<{ [key: string]: any }>;
}

export function EntityFromJSON(json: any): Entity {
  return EntityFromJSONTyped(json, false);
}

export function EntityFromJSONTyped(
  json: any,
  ignoreDiscriminator: boolean
): Entity {
  if (json === undefined || json === null) {
    return json;
  }
  return {
    id: json["id"],
    name: json["name"],
    note: json["note"],
    status: !exists(json, "status") ? undefined : json["status"],
    isToplevel: json["is_toplevel"],
    attrs: json["attrs"],
  };
}

export function EntityToJSON(value?: Entity | null): any {
  if (value === undefined) {
    return undefined;
  }
  if (value === null) {
    return null;
  }
  return {
    name: value.name,
    note: value.note,
    status: value.status,
  };
}
