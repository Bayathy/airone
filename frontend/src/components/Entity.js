import React, {useEffect, useState} from "react";
import Button from "@material-ui/core/Button";
import {Breadcrumbs, Table, TableBody, TableCell, TableContainer, TableHead, TableRow} from "@material-ui/core";
import Paper from "@material-ui/core/Paper";
import {Link} from "react-router-dom";
import EditIcon from '@material-ui/icons/Edit';
import HistoryIcon from '@material-ui/icons/History';
import GroupIcon from '@material-ui/icons/Group';
import DeleteIcon from '@material-ui/icons/Delete';
import {makeStyles} from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";

const useStyles = makeStyles((theme) => ({
    button: {
        margin: theme.spacing(1),
    },
    entityName: {
        margin: theme.spacing(1),
    },
}));

export default function Entity(props) {
    const classes = useStyles();
    const [entities, setEntities] = useState([]);

    useEffect(() => {
        fetch('/entity/api/v1/get_entities')
            .then(resp => resp.json())
            .then(data => setEntities(data.entities));
    }, []);

    return (
        <div className="container-fluid">
            <Breadcrumbs aria-label="breadcrumb">
                <Typography color="textPrimary">エンティティ一覧</Typography>
            </Breadcrumbs>

            <div className="row">
                <div className="col">
                    <div className="float-left">
                        <Button className={classes.button} variant="outlined" color="primary">新規作成</Button>
                        <Button className={classes.button} variant="outlined" color="secondary">エクスポート</Button>
                        <Button className={classes.button} variant="outlined" color="secondary">インポート</Button>
                    </div>
                    <div className="float-right">
                    </div>
                </div>
            </div>

            <TableContainer component={Paper}>
                <Table>
                    <TableHead>
                        <TableRow>
                            <TableCell>
                                <span className={classes.entityName}>エンティティ名</span>
                                <input className={classes.entityName} text='text' placeholder='絞り込む'/>
                            </TableCell>
                            <TableCell><Typography>備考</Typography></TableCell>
                            <TableCell align="right"/>
                        </TableRow>
                    </TableHead>
                    <TableBody>
                        {entities.map((entity) => {
                            return (
                                <TableRow>
                                    <TableCell>
                                        <Typography
                                            component={Link}
                                            to={`/new-ui/entities/${entity.id}`}>
                                            {entity.name}
                                        </Typography>
                                    </TableCell>
                                    <TableCell><Typography>{entity.note}</Typography></TableCell>
                                    <TableCell align="right">
                                        <Button
                                            variant="contained"
                                            color="primary"
                                            className={classes.button}
                                            startIcon={<EditIcon/>}
                                            component={Link}
                                            to={`/entity/edit/${entity.id}`}>
                                            エンティティ編集
                                        </Button>
                                        <Button
                                            variant="contained"
                                            color="primary"
                                            className={classes.button}
                                            startIcon={<HistoryIcon/>}
                                            component={Link}
                                            to={`/entity/history/${entity.id}`}>
                                            変更履歴
                                        </Button>
                                        <Button
                                            variant="contained"
                                            color="primary"
                                            className={classes.button}
                                            startIcon={<GroupIcon/>}
                                            component={Link}
                                            to={`/acl/${entity.id}`}>
                                            ACL
                                        </Button>
                                        <Button
                                            variant="contained"
                                            color="secondary"
                                            className={classes.button}
                                            startIcon={<DeleteIcon/>}
                                            component={Link}
                                            to={`/entity/do_delete/${entity.id}`}>
                                            削除
                                        </Button>
                                    </TableCell>
                                </TableRow>
                            );
                        })}
                    </TableBody>
                </Table>
            </TableContainer>
        </div>
    );
}
