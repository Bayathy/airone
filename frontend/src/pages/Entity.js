import React, {useEffect, useState} from "react";
import Button from "@material-ui/core/Button";
import {Table, TableBody, TableCell, TableContainer, TableHead, TablePagination, TableRow} from "@material-ui/core";
import Paper from "@material-ui/core/Paper";
import {Link} from "react-router-dom";
import EditIcon from '@material-ui/icons/Edit';
import HistoryIcon from '@material-ui/icons/History';
import GroupIcon from '@material-ui/icons/Group';
import DeleteIcon from '@material-ui/icons/Delete';
import {makeStyles} from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import AironeBreadcrumbs from "../components/AironeBreadcrumbs";
import {deleteEntity, getEntities} from "../utils/AironeAPIClient";
import ConfirmableButton from "../components/ConfirmableButton";

const useStyles = makeStyles((theme) => ({
    button: {
        margin: theme.spacing(1),
    },
    entityName: {
        margin: theme.spacing(1),
    },
}));

export default function Entity(props) {
    const classes = useStyles();
    const [entities, setEntities] = useState([]);
    const [page, setPage] = React.useState(0);
    const [rowsPerPage, setRowsPerPage] = React.useState(100);
    const [updated, setUpdated] = useState(false);

    useEffect(() => {
        getEntities()
            .then(resp => resp.json())
            .then(data => setEntities(data.entities))
            .then(_ => setUpdated(false));
    }, [updated]);

    const handleChangePage = (event, newPage) => {
        setPage(newPage);
    };

    const handleChangeRowsPerPage = (event) => {
        setRowsPerPage(+event.target.value);
        setPage(0);
    };

    const handleDelete = (event, entityId) => {
        deleteEntity(entityId)
            .then(_ => setUpdated(true));
    };

    return (
        <div className="container-fluid">
            <AironeBreadcrumbs>
                <Typography component={Link} to='/new-ui/'>Top</Typography>
                <Typography color="textPrimary">エンティティ一覧</Typography>
            </AironeBreadcrumbs>

            <div className="row">
                <div className="col">
                    <div className="float-left">
                        <Button
                            variant="outlined"
                            color="primary"
                            className={classes.button}
                            component={Link}
                            to={`/new-ui/entities/new`}>
                            エンティティ作成
                        </Button>
                        <Button className={classes.button} variant="outlined" color="secondary">エクスポート</Button>
                        <Button className={classes.button} variant="outlined" color="secondary">インポート</Button>
                    </div>
                    <div className="float-right">
                    </div>
                </div>
            </div>

            <Paper>
                <TableContainer component={Paper}>
                    <Table>
                        <TableHead>
                            <TableRow>
                                <TableCell>
                                    <span className={classes.entityName}>エンティティ名</span>
                                    <input className={classes.entityName} text='text' placeholder='絞り込む'/>
                                </TableCell>
                                <TableCell><Typography>備考</Typography></TableCell>
                                <TableCell align="right"/>
                            </TableRow>
                        </TableHead>
                        <TableBody>
                            {entities.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage).map((entity) => {
                                return (
                                    <TableRow>
                                        <TableCell>
                                            <Typography
                                                component={Link}
                                                to={`/new-ui/entities/${entity.id}`}>
                                                {entity.name}
                                            </Typography>
                                        </TableCell>
                                        <TableCell><Typography>{entity.note}</Typography></TableCell>
                                        <TableCell align="right">
                                            <Button
                                                variant="contained"
                                                color="primary"
                                                className={classes.button}
                                                startIcon={<EditIcon/>}
                                                component={Link}
                                                to={`/entity/edit/${entity.id}`}>
                                                エンティティ編集
                                            </Button>
                                            <Button
                                                variant="contained"
                                                color="primary"
                                                className={classes.button}
                                                startIcon={<HistoryIcon/>}
                                                component={Link}
                                                to={`/new-ui/entities/${entity.id}/history`}>
                                                変更履歴
                                            </Button>
                                            <Button
                                                variant="contained"
                                                color="primary"
                                                className={classes.button}
                                                startIcon={<GroupIcon/>}
                                                component={Link}
                                                to={`/new-ui/acl/${entity.id}`}>
                                                ACL
                                            </Button>
                                            <ConfirmableButton
                                                variant="contained"
                                                color="secondary"
                                                className={classes.button}
                                                startIcon={<DeleteIcon/>}
                                                component={Link}
                                                dialogTitle="本当に削除しますか？"
                                                onClickYes={(e) => handleDelete(e, entity.id)}>
                                                削除
                                            </ConfirmableButton>
                                        </TableCell>
                                    </TableRow>
                                );
                            })}
                        </TableBody>
                    </Table>
                </TableContainer>
                <TablePagination
                    rowsPerPageOptions={[100, 250, 1000]}
                    component="div"
                    count={entities.length}
                    rowsPerPage={rowsPerPage}
                    page={page}
                    onChangePage={handleChangePage}
                    onChangeRowsPerPage={handleChangeRowsPerPage}
                />
            </Paper>
        </div>
    );
}
