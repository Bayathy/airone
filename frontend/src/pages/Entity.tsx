import { Box, Typography, Container } from "@mui/material";
import React, { FC } from "react";
import { Link } from "react-router-dom";
import { useAsync } from "react-use";

import { topPath } from "../Routes";
import { AironeBreadcrumbs } from "../components/common/AironeBreadcrumbs";
import { Loading } from "../components/common/Loading";
import { EntityList } from "../components/entity/EntityList";
import { getEntities } from "../utils/AironeAPIClient";

export const Entity: FC = () => {
  const entities = useAsync(async () => {
    const resp = await getEntities();
    const data = await resp.json();
    return data.entities;
  });

  return (
    <Box className="container-fluid">
      <AironeBreadcrumbs>
        <Typography component={Link} to={topPath()}>
          Top
        </Typography>
        <Typography color="textPrimary">エンティティ一覧</Typography>
      </AironeBreadcrumbs>

      <Container maxWidth="lg" sx={{ marginTop: "111px" }}>
        <Box mb="64px">
          <Typography variant="h2" align="center">
            エンティティ一覧
          </Typography>
        </Box>

        {entities.loading ? (
          <Loading />
        ) : (
          <EntityList entities={entities.value} />
        )}
      </Container>
    </Box>
  );
};
