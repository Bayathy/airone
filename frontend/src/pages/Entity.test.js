/**
 * @jest-environment jsdom
 */

import { render, waitFor } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";

import { Entity } from "./Entity";

afterEach(() => {
  jest.clearAllMocks();
});

test("should match snapshot", async () => {
  const entities = [
    {
      id: "1",
      name: "aaa",
      note: "",
    },
    {
      id: "2",
      name: "aaaaa",
      note: "",
    },
    {
      id: "3",
      name: "bbbbb",
      note: "",
    },
  ];

  /* eslint-disable */
  jest
    .spyOn(require("../utils/AironeAPIClient"), "getEntities")
    .mockResolvedValueOnce({
      json() {
        return Promise.resolve({
          entities: entities,
        });
      },
    });
  /* eslint-enable */

  const fragment = await waitFor(() => {
    return render(<Entity />, {
      wrapper: MemoryRouter,
    }).asFragment();
  });

  expect(fragment).toMatchSnapshot();
});
