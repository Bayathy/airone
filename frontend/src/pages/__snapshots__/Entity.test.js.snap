// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-2 container-fluid"
  >
    <nav
      aria-label="breadcrumb"
      class="MuiTypography-root MuiBreadcrumbs-root makeStyles-breadcrumbs-3 MuiTypography-body1 MuiTypography-colorTextSecondary"
    >
      <ol
        class="MuiBreadcrumbs-ol"
      >
        <li
          class="MuiBreadcrumbs-li"
        >
          <a
            class="MuiTypography-root MuiTypography-body1"
            href="/new-ui/"
          >
            Top
          </a>
        </li>
        <li
          aria-hidden="true"
          class="MuiBreadcrumbs-separator"
        >
          /
        </li>
        <li
          class="MuiBreadcrumbs-li"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
          >
            エンティティ一覧
          </p>
        </li>
      </ol>
    </nav>
    <div
      class="MuiBox-root MuiBox-root-4 row"
    >
      <div
        class="MuiBox-root MuiBox-root-5 col"
      >
        <div
          class="MuiBox-root MuiBox-root-6 float-left"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-7 MuiButton-containedPrimary"
            href="/new-ui/entities/new"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                  />
                </svg>
              </span>
              エンティティ作成
            </span>
          </a>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-1 MuiButton-outlinedSecondary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              エクスポート
            </span>
          </button>
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-1 MuiButton-outlinedSecondary"
            href="/new-ui/entities/import"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              インポート
            </span>
          </a>
        </div>
        <div
          class="MuiBox-root MuiBox-root-8 float-right"
        />
      </div>
    </div>
    <div
      class="MuiBox-root MuiBox-root-11"
    >
      <div
        class="MuiBox-root MuiBox-root-12 makeStyles-loading-9"
      >
        <div
          class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style="width: 40px; height: 40px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
        <p
          class="MuiTypography-root makeStyles-text-10 MuiTypography-body1"
        >
          Loading...
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
